src/main/java/fi/vrk/xroad/monitor/extractor/MonitorDataRequestBuilder.java

```diff
@@ -130,14 +134,33 @@ public String getRequestXML(SecurityServerInfo serverInfo) {
         Element bodyElement = document.createElement("SOAP-ENV:Body");
         rootElement.appendChild(bodyElement);
 
-        bodyElement.appendChild(document.createElement("m:getSecurityServerMetrics"));
+        bodyElement.appendChild(metricRequestPayload(document));
 
         document.normalizeDocument();
 
         return getStringFromDocument(document);
     }
 
     /**
+     * Helper function for building metricdata request body.
+     * @param document
+     * @return
+     */
+    private Node metricRequestPayload(Document document) {
+        Element metricRequestRoot = document.createElement("m:getSecurityServerMetrics");
+        if (!queryParameters[0].equals("")) {
+            Element outputSpec = document.createElement("m:outputSpec");
+            Arrays.stream(queryParameters).forEach(parameter -> {
+                outputSpec.appendChild(createElementWithValue(document, "m:outputField", parameter));
+            });
+            metricRequestRoot.appendChild(outputSpec);
+        } else {
+            log.info("No query parameters given, requesting full metric data.");
+        }
+        return metricRequestRoot;
+    }
+
+    /**
```

Comment:
It would be great to have unit test for this new functionality of `MonitorDataRequestBuilder`.

Reply:
Test added